File: 26.py
# 26. Remove Duplicated from Sorted Array
# Reference: https://www.youtube.com/watch?v=DEJAZBq0FDA

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        l = 1  # l is left pointer

        for r in range(1, len(nums)): # r is right pointer
            if nums[r] != nums[r - 1]:
                nums[l] = nums[r]
                l += 1
        
        return l

File: 26.java
// LeetCode 26 Remove Duplicated from Sorted Array
// Reference: https://www.youtube.com/watch?v=zIHe2V5Py3U

class Solution{
    public int removeDuplicates(int[] nums){

        int index = 1;
        for (int i = 0; i < nums.length - 1; i++){
            if (nums[i] != nums[i + 1]){
                nums[index++] = nums[i + 1];
            }
        }
        return index;

    }
}